### PR TITLE
[MNOE-510] Fix confirmation not being sent

### PR DESCRIPTION
**Pull request by x4d3**
I am not entirely sure it is the best solution.

I am a bit confused by the original implementation from devise and I don't see the case where they manage when email_was is nil. 

Original Implementation
```
        def postpone_email_change_until_confirmation_and_regenerate_confirmation_token
          @reconfirmation_required = true
          self.unconfirmed_email = self.email
          self.email = self.email_was
          self.confirmation_token = nil
          generate_confirmation_token
        end

        def postpone_email_change?
          postpone = self.class.reconfirmable && email_changed? && !@bypass_confirmation_postpone && self.email.present?
          @bypass_confirmation_postpone = false
          postpone
        end
```
Without this change, at user creation, on `resource.save`, `postpone_email_change_until_confirmation_and_regenerate_confirmation_token` is called because `postpone_email_change?` is `true`, `resource.email` is set to `nil` and then validation are failing as the `email` is nil.

The error may be due to the workaround implemented for the callback around saving, emulating Active record behaviour. Maybe it's not in the right order.

@ouranos We could discuss it together.

### CODE REVIEW

File: core/app/models/mno_enterprise/base_resource.rb [hunk 1]
```diff
@@ -61,16 +61,16 @@ def update_attribute(name, value)
     # emulate active record call of callbacks
     def save(*args)
       run_callbacks :save do
```
Comment (alachaum) [inline core/app/models/mno_enterprise/base_resource.rb]: We should differentiate between create and update on the `save` method.

See: https://github.com/rails/rails/blob/310918f6a194bf5752fe1025930881756f5d8a8e/activerecord/lib/active_record/persistence.rb#L128

Based on this we should call the `:save` and `:create` or `:update` callbacks

File: core/app/models/mno_enterprise/base_resource.rb [hunk 2]
```diff
@@ -61,16 +61,16 @@ def update_attribute(name, value)
...
-        super()
+        run_callbacks :update do
+          super()
+        end
       end
     end
 
     # emulate active record call of callbacks, a bit different as before_update is called before before_save
     def update_attributes(attrs = {})
       self.attributes = attrs
```
Comment (alachaum) [inline core/app/models/mno_enterprise/base_resource.rb]: We should consider including `ActiveModel::AttributeAssignment` and use the `assign_attributes` method.
https://github.com/rails/rails/blob/master/activemodel/lib/active_model/attribute_assignment.rb

The current method replaces all the attributes and does not use the model custom setters. Not sure it's ideal.

See original implementation of `update_attributes/update` in `ActiveRecord::Persistence`:
https://github.com/rails/rails/blob/310918f6a194bf5752fe1025930881756f5d8a8e/activerecord/lib/active_record/persistence.rb#L280

**Comment by x4d3** [comment]
@alachaum I've adressed your comment regarding the differentiation between update and create.
I created a Jira regarding assign_attributes

**Comment by alachaum** [comment]
WTF the build is red?

**Comment by ouranos** [comment]
I'm taking this

**Comment by ouranos** [comment]
Closing in favor of #473

**Pull request closed without merging**